install-wrappers.sh

```diff
@@ -58,6 +64,45 @@ else
     esac
 fi
 
+if [ -n "${HOST_CLANG-}" ]; then
+    HOST_CLANG_EXE=$(command -v $HOST_CLANG)
+    HOST_CLANG_VER=$(echo "__clang_major__ __clang_minor__ __clang_patchlevel__" | $HOST_CLANG_EXE -E -P -x c - | xargs printf '%d.%d.%d')
+
+    mkdir -p $PREFIX/bin
+
+	# ex. /usr/lib/llvm-17/lib/clang/17
+    resdir=$($HOST_CLANG -print-resource-dir)
+	# ex. /usr/lib/llvm-17
+    llvmdir=${resdir%/lib/clang/*}
+	# ex /lib/clang/17
+    clangres=${resdir#$llvmdir}
+
+    mkdir -p $PREFIX$clangres
+
+    # link the header directory, prevent modification
+    ln -snf $resdir/include $PREFIX$clangres/include
+
+    # Note: clang will detect the "InstalledDir" based on the path that was used to invoke the tools
+    # This might still have some hidden effects
+    printf '#!/bin/sh\nsr=$(dirname "$(dirname "$(readlink -f "$0")")")\nexec %s -resource-dir="$sr"%s --sysroot="$sr" "$@"\n' "$HOST_CLANG_EXE" "$clangres" > $PREFIX/bin/clang
+    # printf '#!/bin/sh\nsr=$(dirname "$(dirname "$(readlink -f "$0")")")\nexec %s -resource-dir="$sr"%s --sysroot="$sr" "$@"\n' "$(readlink -f "$HOST_CLANG_EXE")" "$clangres" > $PREFIX/bin/clang
+    chmod 755 $PREFIX/bin/clang
+    ln -sf clang $PREFIX/bin/clang++
+    ln -sf clang $PREFIX/bin/clang-cpp
```

Comment:
It would be good to set up `clang-$CLANG_MAJOR` as well. If using the native executable form of `clang-target-wrapper`, that's what we're executing and not plain `clang`.

Reply:
Ok.

But TBH I would rather just patch up the wrappers with the path to clang.

Reply:
Bwt. I managed to build via CI, but I am not used to Github: https://github.com/nolange/llvm-mingw/tree/wip

Some pointers there? probably should copy or parameterize linux+asserts and try compiling ffmpeg?

Reply:
Hmm, I guess that'd work as well, but setting up symlinks like this probably is a smaller diff to how things are set up in the default case.

Reply:
Hm, I'm not sure I entirely follow the reasoning here...

Anyway, fair enough, let's drop this.

Yes, the native `clang-target-wrapper` is normally only used on Windows. One could consider to use it on Unix though, if one thinks the runtime overhead of a shell script wrapper matters. And I test running it on Unix occasionally when doing modifications to it, if I don't want to test them on actual Windows at the moment.

Even if using the native `clang-target-wrapper` on Unix, we only direct it to use `clang-$MAJOR` on actual Windows, while it keeps executing `clang` on Unix otherwise, so the point is moot anyway.

So TL;DR, let's drop this issue.

Reply:
The symlinks are fine to build llvm-mingw, but not ideal for distribution and usage (not really nice to shadow `clang`, `llvm-*`).

I am finishing up this PR soon, as I dont have time to invest further (atleast right now).

Current state is here: https://github.com/nolange/llvm-mingw/actions/runs/7535181000

PS. `reasoning`: It should be possible to use a toolchain built via CI on different systems, aslong as the installed clang/llvm is the same major version and the distro is compatible (for example debian-based).